tests/cucumber/acceptance/pageObjects/FilePicker.js

```diff
@@ -0,0 +1,17 @@
+class FilePicker {
+    constructor() {
+        this.fileUploadSelector = 'span.fp-repo-name:has-text("owncloud")';
+        this.fileListSelector ='//div[@class="fp-tableview"]//span[@class="fp-filename"]';
+        this.viewByListFileSelector = 'a[role="button"][title="Display folder with file details"]'
+    }
+
+    async selectRepositoryForUpload(){
```

Comment:
its not for upload

```suggestion
    async selectRepository(){
```

also make the name dynamic